09-rendering/02-TemplateRenderer/components/TemplateRenderer.vue

```diff
@@ -20,5 +20,25 @@ export default {
       default: () => [],
     },
   },
+
+  computed: {
+    dynamicComponent() {
+      const components = this.components;
+      const template = this.template;
+      const bindings = this.$options.props.bindings;
+
+      return defineComponent({
+        props: { bindings },
+        components,
+        render: compile(template),
+      });
```

Comment:
Render function from compiled template may be moved to a separated computed. So the same template won't compile again.